From b8ab64617cbecfaa6c8e56c70fe50746242de65d Mon Sep 17 00:00:00 2001
From: fis <ybbs.daans@hotmail.com>
Date: Sun, 18 Nov 2018 16:39:33 +1300
Subject: [PATCH] Remove `record_property_env'.

---
 googletest/test/gtest_unittest.cc | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/googletest/test/gtest_unittest.cc b/googletest/test/gtest_unittest.cc
index f7213fb..542d6f8 100644
--- a/googletest/test/gtest_unittest.cc
+++ b/googletest/test/gtest_unittest.cc
@@ -2116,10 +2116,6 @@ class UnitTestRecordPropertyTestEnvironment : public Environment {
   }
 };
 
-// This will test property recording outside of any test or test case.
-static Environment* record_property_env =
-    AddGlobalTestEnvironment(new UnitTestRecordPropertyTestEnvironment);
-
 // This group of tests is for predicate assertions (ASSERT_PRED*, etc)
 // of various arities.  They do not attempt to be exhaustive.  Rather,
 // view them as smoke tests that can be easily reviewed and verified.
-- 
2.19.1

