From 0131e6bcc2e4d102c8fba7993a5d6d5aefae5bff Mon Sep 17 00:00:00 2001
From: fis <ybbs.daans@hotmail.com>
Date: Sun, 13 May 2018 05:24:39 +0800
Subject: [PATCH 1/2] Build shared library

---
 CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7676fe1..01705c4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,13 +5,13 @@ project(rabit VERSION 0.0.0)
 option(RABIT_BUILD_TESTS "Build rabit tests" OFF)
 option(RABIT_BUILD_MPI "Build MPI" OFF)
 
-add_library(rabit src/allreduce_base.cc src/allreduce_robust.cc src/engine.cc src/c_api.cc)
-add_library(rabit_base src/allreduce_base.cc src/engine_base.cc src/c_api.cc)
-add_library(rabit_empty src/engine_empty.cc src/c_api.cc)
+add_library(rabit SHARED src/allreduce_base.cc src/allreduce_robust.cc src/engine.cc src/c_api.cc)
+add_library(rabit_base SHARED src/allreduce_base.cc src/engine_base.cc src/c_api.cc)
+add_library(rabit_empty SHARED src/engine_empty.cc src/c_api.cc)
 
 set(rabit_libs rabit rabit_base rabit_empty)
 if(RABIT_BUILD_MPI)
-  add_library(rabit_mpi src/engine_mpi.cc)
+  add_library(rabit_mpi SHARED src/engine_mpi.cc)
   list(APPEND rabit_libs rabit_mpi)
 endif()
 
-- 
2.14.3

