From 6e8b6074d7ccbacc3ee679329581b53afe41af5a Mon Sep 17 00:00:00 2001
From: fis <ybbs.daans@hotmail.com>
Date: Sun, 13 May 2018 17:23:00 +0800
Subject: [PATCH] Fix building shared library.

---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7676fe1..590feb5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,12 +11,14 @@ add_library(rabit_empty src/engine_empty.cc src/c_api.cc)
 
 set(rabit_libs rabit rabit_base rabit_empty)
 if(RABIT_BUILD_MPI)
+  find_package(MPI REQUIRED)
   add_library(rabit_mpi src/engine_mpi.cc)
+  target_link_libraries(rabit_mpi ${MPI_CXX_LIBRARIES})
   list(APPEND rabit_libs rabit_mpi)
 endif()
 
 if(RABIT_BUILD_TESTS)
-  add_library(rabit_mock src/allreduce_base.cc src/allreduce_robust.cc src/engine_mock.cc src/c_api.cc)
+  add_library(rabit_mock STATIC src/allreduce_base.cc src/allreduce_robust.cc src/engine_mock.cc src/c_api.cc)
   list(APPEND rabit_libs rabit_mock) # add to list to apply build settings, then remove
 endif()
 
-- 
2.14.3

