From ca9531a6892a07d5226580392f29febc1092dfb6 Mon Sep 17 00:00:00 2001
From: fis <ybbs.daans@hotmail.com>
Date: Sun, 13 May 2018 05:30:06 +0800
Subject: [PATCH] Link mpi.

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 01705c4..029e3cd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,6 +12,9 @@ add_library(rabit_empty SHARED src/engine_empty.cc src/c_api.cc)
 set(rabit_libs rabit rabit_base rabit_empty)
 if(RABIT_BUILD_MPI)
   add_library(rabit_mpi SHARED src/engine_mpi.cc)
+  find_package(MPI REQUIRED)
+  add_definitions(-DUSE_MPI)
+  target_link_libraries(rabit_mpi ${MPI_CXX_LIBRARIES})
   list(APPEND rabit_libs rabit_mpi)
 endif()
 
-- 
2.14.3

