From b702db629ac1b5540d8dd3e301a235b672d2cbdc Mon Sep 17 00:00:00 2001
From: fis <ybbs.daans@hotmail.com>
Date: Tue, 12 Jun 2018 23:14:49 +0800
Subject: [PATCH] Build test.

---
 CMakeLists.txt | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 99cff60..ee893a8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,6 +9,7 @@ endif ()
 # If building with tests, set flags to generate profile/coverage data
 # for the compilation of the main code.
 include(rct.cmake)
+
 # set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-rtti")
 # set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
 # set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fstack-protector-all")
@@ -17,7 +18,4 @@ include(rct.cmake)
 # set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fstack-protector-all")
 # set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wstack-protector")
 
-# add_executable(dbtest rct/dbtest.cpp)
-# target_link_libraries(dbtest rct)
-
-
+add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/tests)
-- 
2.14.4

