From 33392e7a1b4bac9d3f63aa426c524596988c9265 Mon Sep 17 00:00:00 2001
From: fis <ybbs.daans@hotmail.com>
Date: Sun, 13 May 2018 06:30:49 +0800
Subject: [PATCH] Add install

---
 CMakeLists.txt | 22 +++++++++++++---------
 1 file changed, 13 insertions(+), 9 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a6d0ac40..a4fc828a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,9 +8,9 @@ set_default_configuration_release()
 msvc_use_static_runtime()
 
 # Options
-option(USE_CUDA  "Build with GPU acceleration") 
-option(USE_AVX  "Build with AVX instructions. May not produce identical results due to approximate math." OFF) 
-option(USE_NCCL "Build using NCCL for multi-GPU. Also requires USE_CUDA") 
+option(USE_CUDA  "Build with GPU acceleration")
+option(USE_AVX  "Build with AVX instructions. May not produce identical results due to approximate math." OFF)
+option(USE_NCCL "Build using NCCL for multi-GPU. Also requires USE_CUDA")
 option(JVM_BINDINGS "Build JVM bindings" OFF)
 option(GOOGLE_TEST "Build google tests" OFF)
 option(R_LIB "Build shared library for R package" OFF)
@@ -70,7 +70,7 @@ include_directories (
     # ${PROJECT_SOURCE_DIR}/rabit/include
     )
 
-file(GLOB_RECURSE SOURCES 
+file(GLOB_RECURSE SOURCES
     src/*.cc
     src/*.h
     include/*.h
@@ -114,7 +114,7 @@ if(USE_CUDA)
   cmake_minimum_required(VERSION 3.5)
 
   add_definitions(-DXGBOOST_USE_CUDA)
-  
+
   include_directories(cub)
 
   if(USE_NCCL)
@@ -126,7 +126,7 @@ if(USE_CUDA)
     message("CUDA 9.0 detected, adding Volta compute capability (7.0).")
     set(GPU_COMPUTE_VER "${GPU_COMPUTE_VER};70")
   endif()
-  
+
   set(GENCODE_FLAGS "")
   format_gencode_flags("${GPU_COMPUTE_VER}" GENCODE_FLAGS)
   set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS};--expt-extended-lambda;--expt-relaxed-constexpr;${GENCODE_FLAGS};-lineinfo;")
@@ -139,11 +139,11 @@ if(USE_CUDA)
   endif()
 
   cuda_add_library(gpuxgboost ${CUDA_SOURCES} STATIC)
-  
+
   if(USE_NCCL)
     target_link_libraries(gpuxgboost nccl)
   endif()
-  list(APPEND LINK_LIBRARIES gpuxgboost) 
+  list(APPEND LINK_LIBRARIES gpuxgboost)
 endif()
 
 
@@ -205,7 +205,6 @@ else()
   add_dependencies(xgboost runxgboost)
 endif()
 
-
 # JVM
 if(JVM_BINDINGS)
     find_package(JNI QUIET REQUIRED)
@@ -248,3 +247,8 @@ endif()
 
 # Group sources
 auto_source_group("${SOURCES}")
+
+install(TARGETS xgboost runxgboost
+        RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/bin
+        LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
-- 
2.14.3

