From 396bcf834f18c90707a2c49332c0fe2369e4f92b Mon Sep 17 00:00:00 2001
From: fis <ybbs.daans@hotmail.com>
Date: Sun, 13 May 2018 05:15:11 +0800
Subject: [PATCH] Don't use submodules

---
 CMakeLists.txt | 34 +++++++++++++++++-----------------
 1 file changed, 17 insertions(+), 17 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 22e8c476..a6d0ac40 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,9 +66,9 @@ endif()
 
 include_directories (
     ${PROJECT_SOURCE_DIR}/include
-    ${PROJECT_SOURCE_DIR}/dmlc-core/include
-    ${PROJECT_SOURCE_DIR}/rabit/include
-)
+    # ${PROJECT_SOURCE_DIR}/dmlc-core/include
+    # ${PROJECT_SOURCE_DIR}/rabit/include
+    )
 
 file(GLOB_RECURSE SOURCES 
     src/*.cc
@@ -86,27 +86,27 @@ file(GLOB_RECURSE CUDA_SOURCES
 
 # rabit
 # TODO: Create rabit cmakelists.txt
-set(RABIT_SOURCES
-    rabit/src/allreduce_base.cc
-    rabit/src/allreduce_robust.cc
-    rabit/src/engine.cc
-    rabit/src/c_api.cc
-)
-set(RABIT_EMPTY_SOURCES
-    rabit/src/engine_empty.cc
-    rabit/src/c_api.cc
-)
+# set(RABIT_SOURCES
+#     rabit/src/allreduce_base.cc
+#     rabit/src/allreduce_robust.cc
+#     rabit/src/engine.cc
+#     rabit/src/c_api.cc
+# )
+# set(RABIT_EMPTY_SOURCES
+#     rabit/src/engine_empty.cc
+#     rabit/src/c_api.cc
+# )
 if(MINGW OR R_LIB)
   # build a dummy rabit library
   add_library(rabit STATIC ${RABIT_EMPTY_SOURCES})
-else()
-  add_library(rabit STATIC ${RABIT_SOURCES})
+# else()
+  # add_library(rabit STATIC ${RABIT_SOURCES})
 endif()
 
 
 # dmlc-core
-add_subdirectory(dmlc-core)
-set(LINK_LIBRARIES dmlccore rabit)
+# add_subdirectory(dmlc-core)
+set(LINK_LIBRARIES dmlc rabit)
 
 
 if(USE_CUDA)
-- 
2.14.3

